v2023.12.0-alpha-2 java error

App takes a while to try and launch, then...

```
java.lang.RuntimeException: Exception in Application start method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:893)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Invalid URL: Invalid URL or resource not found
	at javafx.scene.image.Image.validateUrl(Image.java:1133)
	at javafx.scene.image.Image.<init>(Image.java:627)
	at com.faforever.client.fx2j.theme.PersistentNotificationBuilder.build(PersistentNotificationBuilder.java:138)
	at com.faforever.client.fx2j.theme.PersistentNotificationBuilder.build(PersistentNotificationBuilder.java:28)
	at io.github.sheikah45.fx2j.api.Fx2jLoader.load(Fx2jLoader.java:146)
	at com.faforever.client.theme.UiService.loadFxml(UiService.java:73)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:352)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:713)
	at com.faforever.client.theme.UiService$$SpringCGLIB$$0.loadFxml(<generated>)
	at com.faforever.client.notification.PersistentNotificationsController.addNotification(PersistentNotificationsController.java:50)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1116)
	at com.faforever.client.notification.PersistentNotificationsController.onInitialize(PersistentNotificationsController.java:46)
	at com.faforever.client.fx.Controller.initialize(Controller.java:27)
	at com.faforever.client.fx2j.theme.PersistentNotificationsBuilder.build(PersistentNotificationsBuilder.java:69)
	at com.faforever.client.fx2j.theme.headerbar.NotificationButtonBuilder.build(NotificationButtonBuilder.java:66)
	at com.faforever.client.fx2j.theme.headerbar.HeaderBarBuilder.build(HeaderBarBuilder.java:215)
	at com.faforever.client.fx2j.theme.MainBuilder.build(MainBuilder.java:64)
	at com.faforever.client.fx2j.theme.MainBuilder.build(MainBuilder.java:21)
	at io.github.sheikah45.fx2j.api.Fx2jLoader.load(Fx2jLoader.java:146)
	at com.faforever.client.theme.UiService.loadFxml(UiService.java:73)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:352)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:713)
	at com.faforever.client.theme.UiService$$SpringCGLIB$$0.loadFxml(<generated>)
	at com.faforever.client.FafClientApplication.showMainWindow(FafClientApplication.java:87)
	at com.faforever.client.FafClientApplication.start(FafClientApplication.java:72)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:839)
	at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:483)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$10(GtkApplication.java:263)
	... 1 more
Caused by: java.lang.IllegalArgumentException: Invalid URL or resource not found
	at javafx.scene.image.Image.validateUrl(Image.java:1122)
	... 38 more
```

Not sure if it gives any helpful info for you to fix, but I thought I'd share. The last stable version works just fine, so I'm good with that for now of course.

--- a/build.gradle
+++ b/build.gradle
@@ -136,6 +136,7 @@ jacocoTestReport {
 
 test {
   useJUnitPlatform()
+  dependsOn("fx2jJar")
   systemProperties = [
           'junit.jupiter.execution.parallel.enabled': true
   ]

--- a/src/main/java/com/faforever/client/notification/PersistentNotificationController.java
+++ b/src/main/java/com/faforever/client/notification/PersistentNotificationController.java
@@ -1,10 +1,12 @@
 package com.faforever.client.notification;
 
 import com.faforever.client.fx.NodeController;
+import com.faforever.client.theme.ThemeService;
 import javafx.collections.ObservableList;
 import javafx.scene.Node;
 import javafx.scene.control.Button;
 import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
 import javafx.scene.layout.HBox;
 import javafx.scene.layout.Region;
 import lombok.RequiredArgsConstructor;
@@ -26,12 +28,22 @@ public class PersistentNotificationController extends NodeController<Node> {
   private static final String CSS_STYLE_INFO = "info";
   private static final String CSS_STYLE_WARN = "warn";
   private static final String CSS_STYLE_ERROR = "error";
+
   private final NotificationService notificationService;
+  private final ThemeService themeService;
+
   public Node notificationRoot;
   public Label messageLabel;
   public Region icon;
   public HBox actionButtonsContainer;
+  public ImageView closeImage;
+
   private PersistentNotification notification;
+
+  @Override
+  protected void onInitialize() {
+    closeImage.setImage(themeService.getThemeImage("theme/images/small_close.png"));
+  }
 
   /**
    * Sets the notification to display. Populates corresponding UI elements.

--- a/src/main/resources/theme/persistent_notification.fxml
+++ b/src/main/resources/theme/persistent_notification.fxml
@@ -3,7 +3,6 @@
 <?import javafx.geometry.Insets?>
 <?import javafx.scene.control.Button?>
 <?import javafx.scene.control.Label?>
-<?import javafx.scene.image.Image?>
 <?import javafx.scene.image.ImageView?>
 <?import javafx.scene.layout.ColumnConstraints?>
 <?import javafx.scene.layout.GridPane?>
@@ -32,11 +31,7 @@
     </HBox>
       <Button focusTraversable="false" mnemonicParsing="false" onAction="#onCloseButtonClicked" styleClass="icon-button" GridPane.columnIndex="2" GridPane.halignment="RIGHT" GridPane.valignment="CENTER">
       <graphic>
-        <ImageView pickOnBounds="true" preserveRatio="true">
-          <image>
-            <Image url="@images/small_close.png" />
-          </image>
-        </ImageView>
+        <ImageView fx:id="closeImage" pickOnBounds="true" preserveRatio="true"/>
       </graphic>
       <GridPane.margin>
         <Insets left="5.0" />

--- a/src/test/java/com/faforever/client/notification/PersistentNotificationControllerTest.java
+++ b/src/test/java/com/faforever/client/notification/PersistentNotificationControllerTest.java
@@ -1,6 +1,7 @@
 package com.faforever.client.notification;
 
 import com.faforever.client.test.PlatformTest;
+import com.faforever.client.theme.ThemeService;
 import org.hamcrest.CoreMatchers;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
@@ -24,6 +25,8 @@ public class PersistentNotificationControllerTest extends PlatformTest {
 
   @Mock
   private NotificationService notificationService;
+  @Mock
+  private ThemeService themeService;
 
   @BeforeEach
   public void setUp() throws Exception {

